modules/cudacodec/include/opencv2/cudacodec.hpp

```diff
@@ -463,32 +465,42 @@ class CV_EXPORTS_W RawVideoSource
     virtual bool get(const int propertyId, double& propertyVal) const = 0;
 };
 
+/** @brief VideoReader initialization parameters
+@param udpSource Remove validation which can cause VideoReader() to throw exceptions when reading from a UDP source.
+@param liveCapture Prevent delay and eventual disconnection from a live capture source when the rate frames are consumed is less than the rate they are captured.
+Use with caution, frames will be dropped when nextFrame()/grab() are called at a slower rate than the source fps.
+@param minNumDecodeSurfaces Minimum number of internal decode surfaces used by the hardware decoder.  NVDEC will automatically determine the minimum number of
+surfaces it requires for correct functionality and optimal video memory usage but not necessarily for best performance, which depends on the design of the
+overall application. The optimal number of decode surfaces (in terms of performance and memory utilization) should be decided by experimentation for each application,
+but it cannot go below the number determined by NVDEC.
+@param rawMode Allow the raw encoded data which has been read up until the last call to grab() to be retrieved by calling retrieve(rawData,RAW_DATA_IDX).
+*/
+struct CV_EXPORTS_W_SIMPLE VideoReaderInitParams {
+    CV_WRAP VideoReaderInitParams() : udpSource(false), liveSource(false), minNumDecodeSurfaces(0), rawMode(0) {};
+    CV_PROP_RW bool udpSource;
+    CV_PROP_RW bool liveSource;
```

Comment:
I propose to have `allowFrameDrop` everywhere as variable name and point that it's designed for RTSP sreaming. The new name defines concrete functionality, but not abstract mode.